lib/tasks/grade.rake

```diff
@@ -92,6 +97,42 @@ namespace :grade do
 
 end
 
+def sync_specs_with_source
+  reponame = Dir.pwd.split("/").last
+  full_reponame = "appdev-projects/#{reponame}"
+  spec_folder_contents = Octokit.contents(full_reponame, :path => 'spec')
+  spec_subfolder_name = spec_folder_contents.first[:name]
+
+  remote_sha = spec_folder_contents.first[:sha]
+  full_spec_path = "spec/#{spec_subfolder_name}"
+  # Discard unstaged changes in spec folder
+  `git checkout #{full_spec_path} -q`
+  local_sha = `git ls-tree HEAD #{Rails.root.join(full_spec_path)}`.chomp.split[2]
+  
+  unless remote_sha == local_sha
+    spec_files = Octokit.contents(full_reponame, :path => full_spec_path)
+    files = spec_files.map do |file|
```

Comment:
I came across this, which might be faster than looping over each file?

https://stackoverflow.com/a/43545012/18668061